module/documents/actor/actor.mjs

```diff
@@ -166,6 +177,9 @@ export default class Actor5e extends SystemDocumentMixin(Actor) {
 
   /** @inheritDoc */
   prepareDerivedData() {
+    const origin = this.getFlag("dnd5e", "summon.origin");
+    if ( origin ) SummonsData.trackSummon(origin.split(".Item.")[0], this.uuid);
```

Comment:
`parseUuid` might be able to do this a bit more robustly but it might be v12-only so perhaps leave a TODO so we remember to replace this.